cmd/syncthing/main.go

```diff
@@ -559,14 +564,15 @@ func performUpgrade(release upgrade.Release) {
 
 func upgradeViaRest() error {
 	cfg, _ := loadConfig()
-	u, err := url.Parse(cfg.GUI().URL())
+	firstGuiCfg := cfg.GUIs()[0]
+	u, err := url.Parse(firstGuiCfg.URL())
 	if err != nil {
 		return err
 	}
 	u.Path = path.Join(u.Path, "rest/system/upgrade")
 	target := u.String()
 	r, _ := http.NewRequest("POST", target, nil)
-	r.Header.Set("X-API-Key", cfg.GUI().APIKey)
+	r.Header.Set("X-API-Key", firstGuiCfg.APIKey)
```

Comment:
I'd do this as a loop until first success.

Reply:
To be sure sure we're on the same page: should I loop on the whole `upgradeViaRest()` code for each gui (so including request sending and response status code checking) or just the part which forges the request (so up to the `r.Header.Set()` call)?

Reply:
No, try all UI urls, in case some of them are unix sockets for example.

Reply:
So I suggest to add the `// WARNING` here to address the case when we'll have UNIX socket support.

Reply:
Fine